### ISSUE TITLE
java.lang.NoSuchMethodError with 3.2.3

### ISSUE THREAD

**Issue by fnobilia** [salesforce/kafka-junit#56]
I am getting the following error when running with
Kafka JUnit 3.2.3
Kafka kafka_2.13:6.1.1-ccs (note this is leaked by confluent schema-registry 6.1.1)

KafkaJunit 3.2.2 works fine.

For reference, KafkaJUnit is initialised as following
```java
@RegisterExtension
public static final SharedKafkaTestResource kafka = new SharedKafkaTestResource().withBrokers(1);
```

```shell
'scala.collection.GenTraversable scala.collection.immutable.Seq$.empty()'
java.lang.NoSuchMethodError: 'scala.collection.GenTraversable scala.collection.immutable.Seq$.empty()'
	at com.salesforce.kafka.test.KafkaTestServer.start(KafkaTestServer.java:293)
	at com.salesforce.kafka.test.KafkaTestCluster.start(KafkaTestCluster.java:170)
	at com.salesforce.kafka.test.junit5.SharedKafkaTestResource.beforeAll(SharedKafkaTestResource.java:87)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$8(ClassBasedTestDescriptor.java:368)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:368)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:192)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy5.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

In my project, I've regression testing covering from Confluent version `4.0.0` till `6.1.1`. When running these tests with KafkaJunit `3.2.2` they all pass.

For reference, [this is the PR](https://github.com/data-rocks-team/schemaregistry-junit/pull/106) in my project where I am having the issue.

Thank you for your help! =)

**Comment by Crim** [comment]
Ah I wonder if its an issue that I've been testing kafka-junit against [kafka_2.12](https://github.com/salesforce/kafka-junit/blob/master/pom.xml#L90-L91)

Any idea from which version of Kafka is 2.13 recommended over 2.12?  I'll have to dig that up and follow up.

Thanks for the report!

**Comment by Crim** [comment]
I'm having difficulty reproducing this here locally from your project as an example.  I've adjust my POM dependencies like below, removing any explicit inclusions of kafka/kafka-client and replacing it with confluent schema registry.   If I run the tests things seem to build and work fine using kafka 2.7.1

Any idea what I'm missing?

```xml
        <!-- Kafka -->
<!--        <dependency>-->
<!--            <groupId>org.apache.kafka</groupId>-->
<!--            <artifactId>${kafkaScalaVersion}</artifactId>-->
<!--            <version>${kafkaVersion}</version>-->
<!--            <exclusions>-->
<!--                &lt;!&ndash; Don't bring in kafka's logging framework &ndash;&gt;-->
<!--                <exclusion>-->
<!--                    <groupId>org.slf4j</groupId>-->
<!--                    <artifactId>slf4j-log4j12</artifactId>-->
<!--                </exclusion>-->
<!--                <exclusion>-->
<!--                    <groupId>javax.mail</groupId>-->
<!--                    <artifactId>mail</artifactId>-->
<!--                </exclusion>-->
<!--            </exclusions>-->
<!--            <scope>provided</scope>-->
<!--        </dependency>-->
<!--         &lt;!&ndash; Kafka Client&ndash;&gt; -->
<!--                <dependency>-->
<!--                    <groupId>org.apache.kafka</groupId>-->
<!--                    <artifactId>kafka-clients</artifactId>-->
<!--                    <version>${kafkaVersion}</version>-->
<!--                    <scope>provided</scope>-->
<!--                </dependency>-->
        <dependency>
            <groupId>io.confluent</groupId>
            <artifactId>kafka-schema-registry</artifactId>
            <version>6.1.1</version>
        </dependency>
```


Here is the output from `mvn dependency:tree` showing kafka 2.7.1 and scala_kafka2.13:

```
[INFO] ---------------< com.salesforce.kafka.test:kafka-junit5 >---------------
[INFO] Building kafka-junit5 3.2.3                                        [4/4]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ kafka-junit5 ---
[INFO] com.salesforce.kafka.test:kafka-junit5:jar:3.2.3
[INFO] +- com.salesforce.kafka.test:kafka-junit-core:jar:3.2.3:compile
[INFO] +- org.junit.jupiter:junit-jupiter-engine:jar:5.7.2:compile
[INFO] |  +- org.apiguardian:apiguardian-api:jar:1.1.0:compile
[INFO] |  +- org.junit.platform:junit-platform-engine:jar:1.7.2:compile
[INFO] |  |  +- org.opentest4j:opentest4j:jar:1.2.0:compile
[INFO] |  |  \- org.junit.platform:junit-platform-commons:jar:1.7.2:compile
[INFO] |  \- org.junit.jupiter:junit-jupiter-api:jar:5.7.2:compile
[INFO] +- io.confluent:kafka-schema-registry:jar:6.1.1:compile
[INFO] |  +- io.confluent:kafka-schema-registry-client:jar:6.1.1:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.10.5.1:compile
[INFO] |  |  +- jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile
[INFO] |  |  \- org.glassfish.jersey.core:jersey-common:jar:2.31:compile
[INFO] |  |     +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |     +- org.glassfish.hk2:osgi-resource-locator:jar:1.0.3:compile
[INFO] |  |     \- com.sun.activation:jakarta.activation:jar:1.2.2:compile
[INFO] |  +- io.confluent:kafka-json-schema-provider:jar:6.1.1:compile
[INFO] |  |  +- com.github.everit-org.json-schema:org.everit.json.schema:jar:1.12.2:compile
[INFO] |  |  |  +- org.json:json:jar:20201115:compile
[INFO] |  |  |  +- commons-validator:commons-validator:jar:1.6:compile
[INFO] |  |  |  |  +- commons-digester:commons-digester:jar:1.8.1:compile
[INFO] |  |  |  |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  |  |  |  \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] |  |  |  +- com.damnhandy:handy-uri-templates:jar:2.1.8:compile
[INFO] |  |  |  \- com.google.re2j:re2j:jar:1.3:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.10.5:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.10.5:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-joda:jar:2.10.5:compile
[INFO] |  |  |  \- joda-time:joda-time:jar:2.9.9:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.10.5:compile
[INFO] |  |  +- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.10.5:compile
[INFO] |  |  +- com.kjetland:mbknor-jackson-jsonschema_2.13:jar:1.0.39:compile
[INFO] |  |  |  +- org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:jar:1.3.50:compile
[INFO] |  |  |  |  \- org.jetbrains.kotlin:kotlin-scripting-compiler-impl-embeddable:jar:1.3.50:runtime
[INFO] |  |  |  |     +- org.jetbrains.kotlin:kotlin-scripting-common:jar:1.3.50:runtime
[INFO] |  |  |  |     |  \- org.jetbrains.kotlin:kotlin-reflect:jar:1.3.50:runtime
[INFO] |  |  |  |     +- org.jetbrains.kotlin:kotlin-scripting-jvm:jar:1.3.50:runtime
[INFO] |  |  |  |     |  \- org.jetbrains.kotlin:kotlin-script-runtime:jar:1.3.50:runtime
[INFO] |  |  |  |     \- org.jetbrains.kotlinx:kotlinx-coroutines-core:jar:1.1.1:runtime
[INFO] |  |  |  +- javax.validation:validation-api:jar:2.0.1.Final:compile
[INFO] |  |  |  \- io.github.classgraph:classgraph:jar:4.8.21:compile
[INFO] |  |  \- org.jetbrains.kotlin:kotlin-stdlib:jar:1.4.21:compile
[INFO] |  |     +- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.4.21:compile
[INFO] |  |     \- org.jetbrains:annotations:jar:13.0:compile
[INFO] |  +- io.confluent:kafka-protobuf-provider:jar:6.1.1:compile
[INFO] |  |  +- com.squareup.wire:wire-schema:jar:3.6.0:compile
[INFO] |  |  |  +- com.squareup.wire:wire-runtime:jar:3.6.0:compile
[INFO] |  |  |  |  \- org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.4.10:compile
[INFO] |  |  |  |     \- org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.4.10:compile
[INFO] |  |  |  \- com.squareup.okio:okio:jar:2.8.0:compile
[INFO] |  |  +- com.google.protobuf:protobuf-java:jar:3.11.4:compile
[INFO] |  |  \- com.google.protobuf:protobuf-java-util:jar:3.11.4:compile
[INFO] |  |     +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  |     \- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] |  +- org.apache.kafka:kafka_2.13:jar:6.1.1-ccs:compile
[INFO] |  |  +- org.apache.kafka:kafka-raft:jar:6.1.1-ccs:compile
[INFO] |  |  +- com.fasterxml.jackson.module:jackson-module-scala_2.13:jar:2.10.5:compile
[INFO] |  |  |  \- com.fasterxml.jackson.module:jackson-module-paranamer:jar:2.10.5:compile
[INFO] |  |  |     \- com.thoughtworks.paranamer:paranamer:jar:2.8:compile
[INFO] |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-csv:jar:2.10.5:compile
[INFO] |  |  +- net.sf.jopt-simple:jopt-simple:jar:5.0.4:compile
[INFO] |  |  +- com.yammer.metrics:metrics-core:jar:2.2.0:compile
[INFO] |  |  +- org.scala-lang.modules:scala-collection-compat_2.13:jar:2.2.0:compile
[INFO] |  |  +- org.scala-lang.modules:scala-java8-compat_2.13:jar:0.9.1:compile
[INFO] |  |  +- org.scala-lang:scala-library:jar:2.13.3:compile
[INFO] |  |  +- org.scala-lang:scala-reflect:jar:2.13.3:compile
[INFO] |  |  +- com.typesafe.scala-logging:scala-logging_2.13:jar:3.9.2:compile
[INFO] |  |  \- commons-cli:commons-cli:jar:1.4:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:6.1.1-ccs:compile
[INFO] |  |  +- com.github.luben:zstd-jni:jar:1.4.5-6:compile
[INFO] |  |  +- org.lz4:lz4-java:jar:1.7.1:compile
[INFO] |  |  \- org.xerial.snappy:snappy-java:jar:1.1.7.7:compile
[INFO] |  +- org.glassfish.jersey.ext:jersey-bean-validation:jar:2.31:compile
[INFO] |  |  +- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
[INFO] |  |  +- org.glassfish.jersey.core:jersey-server:jar:2.31:compile
[INFO] |  |  |  +- org.glassfish.jersey.core:jersey-client:jar:2.31:compile
[INFO] |  |  |  +- org.glassfish.jersey.media:jersey-media-jaxb:jar:2.31:compile
[INFO] |  |  |  \- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
[INFO] |  |  |     \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
[INFO] |  |  +- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  |  +- jakarta.el:jakarta.el-api:jar:3.0.3:compile
[INFO] |  |  \- org.glassfish:jakarta.el:jar:3.0.3:compile
[INFO] |  +- org.hibernate.validator:hibernate-validator:jar:6.1.7.Final:compile
[INFO] |  |  +- org.jboss.logging:jboss-logging:jar:3.3.2.Final:compile
[INFO] |  |  \- com.fasterxml:classmate:jar:1.3.4:compile
[INFO] |  +- io.confluent:rest-utils:jar:6.1.1:compile
[INFO] |  |  +- org.eclipse.jetty.websocket:javax-websocket-server-impl:jar:9.4.38.v20210224:compile
[INFO] |  |  |  +- org.eclipse.jetty:jetty-annotations:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  +- org.eclipse.jetty:jetty-plus:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  |  \- org.eclipse.jetty:jetty-jndi:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  +- org.eclipse.jetty:jetty-webapp:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  |  \- org.eclipse.jetty:jetty-xml:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  +- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] |  |  |  |  +- org.ow2.asm:asm:jar:9.0:compile
[INFO] |  |  |  |  \- org.ow2.asm:asm-commons:jar:9.0:compile
[INFO] |  |  |  |     +- org.ow2.asm:asm-tree:jar:9.0:compile
[INFO] |  |  |  |     \- org.ow2.asm:asm-analysis:jar:9.0:compile
[INFO] |  |  |  +- org.eclipse.jetty.websocket:javax-websocket-client-impl:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  +- org.eclipse.jetty.websocket:websocket-client:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  |  \- org.eclipse.jetty:jetty-client:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  \- javax.websocket:javax.websocket-client-api:jar:1.0:compile
[INFO] |  |  |  +- org.eclipse.jetty.websocket:websocket-server:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  +- org.eclipse.jetty.websocket:websocket-common:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  |  \- org.eclipse.jetty.websocket:websocket-api:jar:9.4.38.v20210224:compile
[INFO] |  |  |  |  \- org.eclipse.jetty.websocket:websocket-servlet:jar:9.4.38.v20210224:compile
[INFO] |  |  |  \- javax.websocket:javax.websocket-api:jar:1.0:compile
[INFO] |  |  +- org.glassfish.jersey.containers:jersey-container-servlet:jar:2.31:compile
[INFO] |  |  |  \- org.glassfish.jersey.containers:jersey-container-servlet-core:jar:2.31:compile
[INFO] |  |  +- org.glassfish.jersey.inject:jersey-hk2:jar:2.31:compile
[INFO] |  |  |  \- org.glassfish.hk2:hk2-locator:jar:2.6.1:compile
[INFO] |  |  |     +- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.6.1:compile
[INFO] |  |  |     +- org.glassfish.hk2:hk2-api:jar:2.6.1:compile
[INFO] |  |  |     \- org.glassfish.hk2:hk2-utils:jar:2.6.1:compile
[INFO] |  |  +- javax.activation:activation:jar:1.1.1:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-jmx:jar:9.4.38.v20210224:compile
[INFO] |  |  |  \- org.eclipse.jetty:jetty-util:jar:9.4.38.v20210224:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-server:jar:9.4.38.v20210224:compile
[INFO] |  |  |  +- javax.servlet:javax.servlet-api:jar:3.1.0:compile
[INFO] |  |  |  +- org.eclipse.jetty:jetty-http:jar:9.4.38.v20210224:compile
[INFO] |  |  |  \- org.eclipse.jetty:jetty-io:jar:9.4.38.v20210224:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-servlet:jar:9.4.38.v20210224:compile
[INFO] |  |  |  +- org.eclipse.jetty:jetty-security:jar:9.4.38.v20210224:compile
[INFO] |  |  |  \- org.eclipse.jetty:jetty-util-ajax:jar:9.4.38.v20210224:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-servlets:jar:9.4.38.v20210224:compile
[INFO] |  |  |  \- org.eclipse.jetty:jetty-continuation:jar:9.4.38.v20210224:compile
[INFO] |  |  +- org.eclipse.jetty:jetty-jaas:jar:9.4.38.v20210224:compile
[INFO] |  |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.10.5:compile
[INFO] |  |  |  \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.10.5:compile
[INFO] |  |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.10.5:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.5:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.30:compile
[INFO] |  +- io.confluent:confluent-log4j:jar:1.2.17-cp2:compile
[INFO] |  +- org.apache.avro:avro:jar:1.9.2:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.10.2:compile
[INFO] |  |  \- org.apache.commons:commons-compress:jar:1.19:compile
[INFO] |  +- org.apache.zookeeper:zookeeper:jar:3.5.8:compile
[INFO] |  |  +- org.apache.zookeeper:zookeeper-jute:jar:3.5.8:compile
[INFO] |  |  +- org.apache.yetus:audience-annotations:jar:0.5.0:compile
[INFO] |  |  +- io.netty:netty-handler:jar:4.1.48.Final:compile
[INFO] |  |  |  +- io.netty:netty-common:jar:4.1.48.Final:compile
[INFO] |  |  |  +- io.netty:netty-resolver:jar:4.1.48.Final:compile
[INFO] |  |  |  +- io.netty:netty-buffer:jar:4.1.48.Final:compile
[INFO] |  |  |  +- io.netty:netty-transport:jar:4.1.48.Final:compile
[INFO] |  |  |  \- io.netty:netty-codec:jar:4.1.48.Final:compile
[INFO] |  |  \- io.netty:netty-transport-native-epoll:jar:4.1.48.Final:compile
[INFO] |  |     \- io.netty:netty-transport-native-unix-common:jar:4.1.48.Final:compile
[INFO] |  +- com.101tec:zkclient:jar:0.11:compile
[INFO] |  +- io.swagger:swagger-annotations:jar:1.6.2:compile
[INFO] |  +- io.swagger:swagger-core:jar:1.6.2:compile
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.2.1:compile
[INFO] |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.11.1:compile
[INFO] |  |  |  \- org.yaml:snakeyaml:jar:1.26:compile
[INFO] |  |  \- io.swagger:swagger-models:jar:1.6.2:compile
[INFO] |  \- io.confluent:common-utils:jar:6.1.1:compile
[INFO] +- org.apache.curator:curator-test:jar:2.12.0:compile
[INFO] |  +- org.javassist:javassist:jar:3.18.1-GA:compile
[INFO] |  \- com.google.guava:guava:jar:16.0.1:compile
[INFO] +- org.slf4j:slf4j-simple:jar:1.7.30:test
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO] \- javax.xml.bind:jaxb-api:jar:2.3.1:test
[INFO]    \- javax.activation:javax.activation-api:jar:1.2.0:test
```

**Comment by fnobilia** [comment]
I looked into the dependencies and I cannot see anything wrong. [Here](https://gist.github.com/fnobilia/2cbb4961d6927867bb2644825d7368a3) you can find the full list.

I've also tried to bump both KafkaJunit and ConfluentSchemaRegistry to the latest. In this case, the core tests work fine but all regression tests fail.

Do you think the issue could be caused by the new code where you instantiate the KafkaServer via reflection? If so, what could cause different reflection behaviour? I've double-checked the scala version and it matches with yours.

**Comment by fnobilia** [comment]
Ok, I think I found something more! In Scala [2.12.14](https://www.scala-lang.org/api/2.12.14/scala/collection/immutable/Seq.html) the `empty()` function did not exist. While, Scala [2.13.0](https://www.scala-lang.org/api/2.13.0/scala/collection/immutable/Seq.html) offers `empty()`.

I guess that my code is picking up the wrong Scala version for some reason. I don't know at the moment.

The following command shows that my intuition is wrong
```shell
./gradlew :schemaregistry-junit-core:dependencyInsight --dependency org.scala-lang:scala-library 

> Task :schemaregistry-junit-core:dependencyInsight
org.scala-lang:scala-library:2.13.3
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.version         = 8
   ]
   Selection reasons:
      - By conflict resolution : between versions 2.13.3, 2.13.1 and 2.13.0

org.scala-lang:scala-library:2.13.3
+--- com.fasterxml.jackson.module:jackson-module-scala_2.13:2.10.5
|    \--- org.apache.kafka:kafka_2.13:6.1.1-ccs
|         \--- io.confluent:kafka-schema-registry:6.1.1
|              \--- compileClasspath
+--- org.apache.kafka:kafka_2.13:6.1.1-ccs (*)
+--- org.scala-lang:scala-reflect:2.13.3
|    +--- org.apache.kafka:kafka_2.13:6.1.1-ccs (*)
|    \--- com.typesafe.scala-logging:scala-logging_2.13:3.9.2 (requested org.scala-lang:scala-reflect:2.13.0)
|         \--- org.apache.kafka:kafka_2.13:6.1.1-ccs (*)
\--- org.scala-lang.modules:scala-collection-compat_2.13:2.2.0
     \--- org.apache.kafka:kafka_2.13:6.1.1-ccs (*)

org.scala-lang:scala-library:2.13.0 -> 2.13.3
+--- com.typesafe.scala-logging:scala-logging_2.13:3.9.2
|    \--- org.apache.kafka:kafka_2.13:6.1.1-ccs
|         \--- io.confluent:kafka-schema-registry:6.1.1
|              \--- compileClasspath
\--- org.scala-lang.modules:scala-java8-compat_2.13:0.9.1
     \--- org.apache.kafka:kafka_2.13:6.1.1-ccs (*)

org.scala-lang:scala-library:2.13.1 -> 2.13.3
\--- com.kjetland:mbknor-jackson-jsonschema_2.13:1.0.39
     \--- io.confluent:kafka-json-schema-provider:6.1.1
          \--- io.confluent:kafka-schema-registry:6.1.1
               \--- compileClasspath

(*) - dependencies omitted (listed previously)

```

I'll run some more investigations.

**Comment by fnobilia** [comment]
I double-checked everything and I cannot see why the code should use a different scala version from the one reported by gradle.

I've also added a breakpoint at [com.salesforce.kafka.test.KafkaTestServer#L292](https://github.com/salesforce/kafka-junit/blob/5e128ac684cb34da246e6bf438cc776fc528bd84/kafka-junit-core/src/main/java/com/salesforce/kafka/test/KafkaTestServer.java#L292) and used the evaluator inside IntelliJ to make sure I can create the empty Seq and I can create the broker. But when the test run, it keeps failing.

I am running out of ideas here. But I think the problem is more related to my project than KafkaJunit.

If you don't mind, I would like to run some more investigations before closing the issue. =)

**Comment by fnobilia** [comment]
I've run some more tests and found something interesting. I've created a test like this one
```java
package com.datarocks.schemaregistry.test;

import kafka.metrics.KafkaMetricsReporter;
import org.junit.jupiter.api.Test;

class ScalaTestSeqEmpty {

  @Test
  void scalaTestSeqEmpty() {
    System.out.println(scala.collection.immutable.Seq$.MODULE$.<KafkaMetricsReporter>empty());
  }
}
```

The test works. The `empty` function can be found in `Seq`. But the test using KafkaJunit does not pass.

**Comment by fnobilia** [comment]
I did some more tests and refactored my test strategy to make sure each classpath only had one confluent version at a time. KafkaJunit `3.2.3` works with all confluent versions from `4.0.3` to `6.2.0`, except for `6.0.1` and `6.1.1`.

I can conclude that there is a dependency present only in `6.0.1` and `6.1.1` that alongside my dependencies is confusing Gradle and force it to load Scala `12`. Scala `12` in conjunction with a Kafka version before `2.13` creates the issue mentioned in the first message.

I've now forced KafkaJunit `3.2.2` in test running against `6.0.1` and `6.1.1`, using the following approach

```java
testImplementation('com.salesforce.kafka.test:kafka-junit5') {
  version {
    strictly '3.2.2'
  }
}
```

At this point, I don't think there is any problem with KafkaJunit and I will close the issue.

@Crim Thank you for your help!
